src/GameLogic/DataStructures/Entities.ts

```diff
@@ -0,0 +1,33 @@
+import {EntityType} from '../Enums';
+import {Entity} from '../Entity';
+import {Protagonist} from '../Entities/Protagonist';
+
+
+export class Entities {
+	private _entities: Entity[];
+
+	constructor(entities: Entity[] = []) {
+		this._entities = Array.from(entities);
+	}
+
+	public push(entity: Entity): void {
```

Comment:
Can you rename it to `addEntity`? `push` is very array specific with a clearly defined purpose (add at the end of an array), but in this case we really only care about adding the entity to the list of entities, and are not interested on how it happens internally, since the internals are not really exposed.

Reply:
I took the liberty of doing it myself and I merged the PR, since I wanted to work with this PR in place and wanted to clean something up first